api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -437,34 +416,27 @@ private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewLis
     if (Strings.isNullOrEmpty(request.getEtag())) {
       throw new BadRequestException("missing required update field 'etag'");
     }
+    final Timestamp now = new Timestamp(clock.instant().toEpochMilli());
     DbWorkspace workspace =
         workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
             workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
-
+    request.setWorkspaceId(workspace.getWorkspaceId());
     DbDataset dbDataSet = dataSetService.getDbDataSet(workspace, dataSetId).get();
 
     int version = Etags.toVersion(request.getEtag());
     if (dbDataSet.getVersion() != version) {
       throw new ConflictException("Attempted to modify outdated data set version");
     }
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    dbDataSet.setName(request.getName());
-    dbDataSet.setDescription(request.getDescription());
-    dbDataSet.setLastModifiedTime(now);
-    if (request.getDomainValuePairs() != null) {
-      dbDataSet.setIncludesAllParticipants(request.getIncludesAllParticipants());
-      dbDataSet.setCohortIds(request.getCohortIds());
-      dbDataSet.setConceptSetIds(request.getConceptSetIds());
-      dbDataSet.setPrePackagedConceptSetEnum(request.getPrePackagedConceptSet());
-      dbDataSet.setValues(
-          request.getDomainValuePairs().stream()
-              .map(this::getDataSetValuesFromDomainValueSet)
-              .collect(Collectors.toList()));
-    }
+    DbDataset dbMappingConvert = dataSetMapper.dataSetRequestToDb(request);
+    dbMappingConvert.setDataSetId(dbDataSet.getDataSetId());
+    dbMappingConvert.setInvalid(false);
+    dbMappingConvert.setCreatorId(dbDataSet.getCreatorId());
+    dbMappingConvert.setCreationTime(dbDataSet.getCreationTime());
+    dbMappingConvert.setLastModifiedTime(now);
```

Comment:
I think you need to revert this block and leave the code the way it was for purposes of how the rename has to work. This will be a breaking change to the rename dataset.

Reply:
We need to add a test case for the dataset rename that would have caught this type of code change.

Reply:
Is this because rename just going to send the name and id and not the cohort/Concept Ids ?

Reply:
Correct. And this was the fix... so we didn't need a new endpoint.